api/utils/kubernetes.go

```diff
@@ -0,0 +1,48 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+func ParsePodContainerStatuses(containerStatuses []v1.ContainerStatus) (string, string, error) {
```

Comment:
Probably some comment here would be good or rather the error is the error while executing this function instead of the implicit meaning, as its normally expected 
`"", "". err` or `"something", "something", nil`